R/git-default-branch.R

```diff
@@ -0,0 +1,408 @@
+#' Infer default Git branch
+#'
+#' @description
+
+#' Figure out the default branch of the current Git repo, preferably from local
+#' information, but possibly from the `HEAD` ref of the `upstream` or `origin`
+#' remote. Since "default branch" is not a well-defined Git concept, certain
+#' assumptions are baked into this function:
+
+#' * The local Git repo accurately reflects the branch conventions of the
+#' overall project, either by definition (because you control the project) or
+#' because you're keeping up-to-date with the source repository.
+
+#' * Typical default branch names are `main`, `master`, and `default`, which we
+#' look for in that order.
+
+#' * The remote names `origin` and `upstream` are used in the conventional
+#' manner.
+
+#'
+#' If you want to discover and adapt to a branch move, such as a switch from
+#' `master` to `main`, see [use_git_default_branch()] instead. That function
+#' also helps you make such a switch in a project you control.
+#' `git_default_branch()` is a passive function that takes things at face value,
+#' whereas [use_git_default_branch()] makes an active effort to discover or
+#' enact or change.
+#'
+#' @return A branch name
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' git_default_branch()
+#' }
+git_default_branch <- function() {
```

Comment:
This function pre-existed. I just moved its definition from one file to another and renamed it (well, the old version still works but generates a deprecation warning).

I originally named it `git_branch_default()`, out of deference to all the `gert::git_branch_*()` functions. But now I think it makes more sense to use the more natural `git_default_branch()`, in the usethis context.

This needs to somehow be integrated or refactored in light of new default branch detection use cases. The tricky part is that sometimes we want to infer default branch in a very naive and lightweight way and sometimes we want to be skeptical and actively check for change in remote places.